TypeCobol.LanguageServer/Workspace.cs

```diff
@@ -951,6 +952,192 @@ private void FinalCompilationStepCompleted(object cUnit, ProgramClassEvent progr
+                    }
+                    else if (dataDefinition.CodeElement is DataDescriptionEntry dataDescriptionEntry)
+                    {
+                        if (dataDefinition.Picture != null)
+                        {
+                            row.Append(string.Format(PIC, dataDefinition.Picture.Value));
+                        }
+                        string usage = dataDescriptionEntry.Usage?.Token.Text;
+                        if (!string.IsNullOrWhiteSpace(usage))
+                        {
+                            AppendSpaceIfNeeded();
+                            row.Append(usage);
+                        }
+
+                        if (row.Length == initialRowLength && dataDefinition.ChildrenCount > 0)
+                        {
+                            row.Append(GROUP);
+                        }
+                    }
+
+                    if (dataDefinition.IsTableOccurence)
+                    {
+                        AppendSpaceIfNeeded();
+                        row.Append(string.Format(OCCURS, dataDefinition.MaxOccurencesCount));
+                    }
+
+                    row.Append(separator);
+
+                    void AppendSpaceIfNeeded()
+                    {
+                        if (row.Length != initialRowLength)
+                        {
+                            row.Append(SPACE);
+                        }
+                    }
+                }
+
+                static object GetEnd(long start, long length)
+                {
+                    var end = start + length - 1;
+                    return (end > 0) ? end : UNDEFINED;
+                }
+            }
+        }
+
+        private List<Tuple<int, DataDefinition>> CollectDataLayoutNodes(CompilationUnit compilationUnit)
+        {
+            var result = new List<Tuple<int, DataDefinition>>();
+            Node dataDivision = compilationUnit.TemporaryProgramClassDocumentSnapshot.Root.MainProgram.GetChildren<DataDivision>().FirstOrDefault();
+            if (dataDivision != null)
+            {
+                // Consider data declared in the Working storage
+                var workingStorage = dataDivision.GetChildren<WorkingStorageSection>().FirstOrDefault();
+                if (workingStorage != null)
+                {
+                    result.AddRange(CollectDataLayoutNodes(0, workingStorage));
+                }
+
+                // Consider also data declared in the Local storage
+                var localStorage = dataDivision.GetChildren<LocalStorageSection>().FirstOrDefault();
+                if (localStorage != null)
+                {
+                    result.AddRange(CollectDataLayoutNodes(0, localStorage));
+                }
+            }
+
+            return result;
+
+            static IEnumerable<Tuple<int, DataDefinition>> CollectDataLayoutNodes(int nodeLevel, Node node)
```

Comment:
For performance reason, it's better to pass a list as argument and then the method will fill the list.
Rather than using `yield return`.

Reply:
Done